docs/client-concepts/high-level/indexing/pipelines.asciidoc

```diff
@@ -0,0 +1,134 @@
+:ref_current: https://www.elastic.co/guide/en/elasticsearch/reference/6.x
+
+:github: https://github.com/elastic/elasticsearch-net
+
+:nuget: https://www.nuget.org/packages
+
+////
+IMPORTANT NOTE
+==============
+This file has been generated from https://github.com/elastic/elasticsearch-net/tree/6.x/src/Tests/Tests/ClientConcepts/HighLevel/Indexing/Pipelines.doc.cs. 
+If you wish to submit a PR for any spelling mistakes, typos or grammatical errors for this file,
+please modify the original csharp file found at the link and submit the PR with that change. Thanks!
+////
+
+[[ingest-pipelines]]
+=== Ingest Pipelines
+
+Elasticsearch will automatically re-route index requests to ingest nodes,
+however with some careful consideration you can optimise this path.
+
+[source,csharp]
+----
+public class Person
+{
+    public int Id { get; set; }
+    public string FirstName { get; set; }
+    public string LastName { get; set; }
+    public string IpAddress { get; set; }
+    public GeoIp GeoIp { get; set; }
+}
+
+public class GeoIp
+{
+    public string CityName { get; set; }
+    public string ContinentName { get; set; }
+    public string CountryIsoCode { get; set; }
+    public GeoLocation Location { get; set; }
+    public string RegionName { get; set; }
+}
+----
+
+==== Create an ingestion pipeline
+
+Assuming we are indexing Person documents, we can create an ingestion pipeline that manipulates the
+incoming values before they are indexed.
+
+Lets assume that our application always expects surnames to be capitalised, and for initials to
+be indexed into their own field. We also have an IP address that we'd like to convert into
+a human-readable location.
+
+We could achieve this requirement by creating a custom mapping and creating an ingest pipeline.
+The Person type can then be used as-is, without making any changes.
+
+[source,csharp]
+----
+client.CreateIndex("people");
+
+client.Map<Person>(p => p
+    .Index("people")
+    .AutoMap() <1>
+    .Properties(props => props
+        .Keyword(t => t.Name("initials")) <2>
+        .Ip(t => t.Name(dv => dv.IpAddress)) <3>
+        .Object<GeoIp>(t => t.Name(dv => dv.GeoIp)) <4>
+    )
+);
+
+client.PutPipeline("person-pipeline", p => p
+    .Processors(ps => ps
+        .Uppercase<Person>(s => s
+            .Field(t => t.LastName) <5>
+        )
+        .Script(s => s
+            .Lang("painless") <6>
+            .Source("ctx.initials = ctx.firstName.substring(0,1) + ctx.lastName.substring(0,1)")
+        )
+        .GeoIp<GeoIp>(s => s <7>
+            .Field("ipAddress")
+            .TargetField("geoIp")
```

Comment:
lambda expressions?

Reply:
Not possible as lambdas target `GeoIp` type and not `Person` type.

Reply:
The `T` in `.GeoIp<T>` is only used for strongly typed access for the `.Field()` and `.TargetField()` methods, so can be `Person` in this case. 

The `GeoIp` type will also need `DataMemberAttribute` as the casing of the properties produced by `ingest-geoip` ar snake-cased